win32/devdoc/gballoc_ll_jemalloc.md

```diff
@@ -165,4 +167,28 @@ static void jemalloc_print_stats_callback(void* context, const char* text)
 
 **SRS_GBALLOC_LL_JEMALLOC_01_009: [** If `text` is NULL, `jemalloc_print_stats_callback` shall return. **]**
 
-**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
\ No newline at end of file
+**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
+
+### gballoc_ll_set_decay
+
+```c
+MOCKABLE_FUNCTION(, int, gballoc_ll_set_decay, int64_t, decay_milliseconds);
+```
+
+`gballoc_ll_set_decay` sets the dirty and muzzy pages decay time to `decay_milliseconds` milliseconds.
+
+**SRS_GBALLOC_LL_JEMALLOC_28_001: [** `gballoc_ll_set_decay` shall advance jemalloc's internal epoch counter. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_002: [** `gballoc_ll_set_decay` shall set the dirty decay time for new arenas to `decay_milliseconds` milliseconds. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_003: [** `gballoc_ll_set_decay` shall set the muzzy decay time for new arenas to `decay_milliseconds` milliseconds. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_004: [** `gballoc_ll_set_decay` shall fetch the number of existing jemalloc arenas. **]**
+
+**SRS_GBALLOC_LL_JEMALLOC_28_005: [** For each existing arena: **]**
```

Comment:
From what I understand, arenas are never deleted. I am still looking for the documentation that can confirm this or maybe will try to dig into the code further. So in that case we would not need a locking mechanism since new arenas would already have their decay set, and existing arenas would be updated in this loop.